app/com/lynxanalytics/sandbox/ScalaScript.scala

```diff
@@ -293,15 +264,12 @@ object ScalaScript {
     })
   }
 
-  private def withContextClassLoader[T](func: => T): T = synchronized {
-    // IMAIN.compile changes the class loader and does not restore it.
-    // https://issues.scala-lang.org/browse/SI-8521
-    if (engine == null) engine = createEngine()
-    val cl = Thread.currentThread().getContextClassLoader
+  private def withEngine[T](func: => T): T = synchronized {
+    engine = createEngine()
```

Comment:
Please check if this makes things very slow.

Reply:
> Please check if this makes things very slow.

Looks like it's okay. I've measured getting the histogram of `ordinal % 10` and it's around 1 second per million vertices both before and after.